### ISSUE TITLE
CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!

### ISSUE THREAD

**Issue by Toni981** [home-assistant/supervisor#1819]
**Home Assistant release with the issue:**
0.112.2

**Operating environment (HassOS/Generic):**
HassOS

**Supervisor logs:**

CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!

**Description of problem:**

Good evening.
I am trying to update version 0.112.2 to version 0.112.3 but I have the error:

CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!

I have also tried from Terminal but I have the same error.

A few days ago I reprogrammed everything from scratch so I am sure that I have no problems in the few addons that I have. Can you give me some suggestions?

It has already happened to me before going from 0.112.0 to 0.112.2 where I had to install everything from scratch but now I find myself with the same problem ...

**Comment by balloob** [comment]
Please fill in the issue template.

**Comment by Toni981** [comment]
ok sorry...I have correct it. can you help me?

**Comment by pvizeli** [comment]
There should be more inside the log as this one line

**Comment by fedebruni84** [comment]
> There should be more inside the log as this one line

Look at this, seems there's no other lines in the log:
https://community.home-assistant.io/t/error-while-updating-from-0-111-4-to-0-112-3/210086

**Comment by Toni981** [comment]
Hi, yesterday I downloaded 0.112.3 and installed the backup. 
Today I saw the update to version 0.112.4 and I tried to do the update but I have this problem:

`20-07-09 09:05:12 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository
20-07-09 09:05:12 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/danielwelch/hassio-addons repository
20-07-09 09:05:12 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/hassio-addons repository
20-07-09 09:05:12 INFO (MainThread) [supervisor.updater] Fetch update data from https://version.home-assistant.io/stable.json
20-07-09 09:05:14 INFO (MainThread) [supervisor.store] Load add-ons from store: 66 all - 0 new - 0 remove
20-07-09 09:06:17 INFO (MainThread) [supervisor.homeassistant] Update Home Assistant to version 0.112.4
20-07-09 09:06:17 INFO (SyncWorker_16) [supervisor.docker.interface] Update image homeassistant/raspberrypi4-homeassistant:0.112.3 to homeassistant/raspberrypi4-homeassistant:0.112.4
20-07-09 09:06:17 INFO (SyncWorker_16) [supervisor.docker.interface] Pull image homeassistant/raspberrypi4-homeassistant tag 0.112.4.
20-07-09 09:16:35 INFO (SyncWorker_16) [supervisor.docker.interface] Stop homeassistant application
20-07-09 09:17:36 INFO (SyncWorker_16) [supervisor.docker.interface] Clean homeassistant application
20-07-09 09:17:36 INFO (MainThread) [supervisor.homeassistant] Update pulse/client.config: /data/tmp/homeassistant_pulse
20-07-09 09:17:38 INFO (SyncWorker_10) [supervisor.docker.homeassistant] Start homeassistant homeassistant/raspberrypi4-homeassistant with version 0.112.4
20-07-09 09:17:53 INFO (MainThread) [supervisor.homeassistant] Updated Home Assistant API token
20-07-09 09:17:53 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation in progress
20-07-09 09:20:23 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation done
20-07-09 09:30:28 WARNING (MainThread) [supervisor.homeassistant] Don't wait anymore of Home Assistant startup!
20-07-09 09:30:28 CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!
20-07-09 09:30:28 INFO (MainThread) [supervisor.homeassistant] Update Home Assistant to version 0.112.3
20-07-09 09:30:28 INFO (SyncWorker_3) [supervisor.docker.interface] Update image homeassistant/raspberrypi4-homeassistant:0.112.4 to homeassistant/raspberrypi4-homeassistant:0.112.3
20-07-09 09:30:28 INFO (SyncWorker_3) [supervisor.docker.interface] Pull image homeassistant/raspberrypi4-homeassistant tag 0.112.3.
20-07-09 09:30:30 INFO (SyncWorker_3) [supervisor.docker.interface] Stop homeassistant application
20-07-09 09:30:34 INFO (SyncWorker_3) [supervisor.docker.interface] Clean homeassistant application
20-07-09 09:30:34 INFO (MainThread) [supervisor.homeassistant] Update pulse/client.config: /data/tmp/homeassistant_pulse
20-07-09 09:30:40 INFO (SyncWorker_15) [supervisor.docker.homeassistant] Start homeassistant homeassistant/raspberrypi4-homeassistant with version 0.112.3
20-07-09 09:30:55 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation in progress`

**Comment by giu1io** [comment]
I have the same problem going from 0.111.4 to 0.112.3.
Haven’t found a solution at the moment.

**Comment by Toni981** [comment]
For version 0.112.0 to version 0.112.2 I solved it by downloading the file from the site and installing the backup. 
But now from version 0.112.2 to version 0.112.3 it gives me error even installing the backup .. I don't know what to do anymore since I don't have many things stored.

In addition to the addons put on the screen I also have Hacs (to use shelly and sonoff)

There will definitely be a solution ...

**Comment by Toni981** [comment]
who can help me?

**Comment by foxy82** [comment]
I have the same issue as well my logs don't give much help - although could it be the migration to the new DB format meaning it is taking too long to start so rolls back? Can I manually update the homeassistant docker from the command line or will that cause issues?

`
20-07-20 11:08:59 INFO (MainThread) [supervisor.homeassistant] Update Home Assistant to version 0.112.4
20-07-20 11:08:59 INFO (SyncWorker_3) [supervisor.docker.interface] Update image homeassistant/qemux86-64-homeassistant:0.111.3 to homeassistant/qemux86-64-homeassistant:0.112.4
20-07-20 11:08:59 INFO (SyncWorker_3) [supervisor.docker.interface] Pull image homeassistant/qemux86-64-homeassistant tag 0.112.4.
20-07-20 11:10:14 INFO (SyncWorker_3) [supervisor.docker.interface] Stop homeassistant application
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:1000 is not str
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:1000 is not str
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:1000 is not str
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
20-07-20 11:10:15 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
20-07-20 11:10:21 INFO (SyncWorker_3) [supervisor.docker.interface] Clean homeassistant application
20-07-20 11:10:26 INFO (MainThread) [supervisor.homeassistant] Update pulse/client.config: /data/tmp/homeassistant_pulse
20-07-20 11:10:29 INFO (SyncWorker_1) [supervisor.docker.homeassistant] Start homeassistant homeassistant/qemux86-64-homeassistant with version 0.112.4
20-07-20 11:10:41 INFO (MainThread) [supervisor.auth] Auth request from core_mosquitto for iot
20-07-20 11:10:42 INFO (MainThread) [supervisor.auth] Home Assistant not running, check cache
20-07-20 11:10:42 INFO (MainThread) [supervisor.auth] Cache hit for iot
20-07-20 11:10:45 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation in progress
20-07-20 11:11:20 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation done
20-07-20 11:21:24 WARNING (MainThread) [supervisor.homeassistant] Don't wait anymore of Home Assistant startup!
20-07-20 11:21:24 CRITICAL (MainThread) [supervisor.homeassistant] HomeAssistant update fails -> rollback!
20-07-20 11:21:24 INFO (MainThread) [supervisor.homeassistant] Update Home Assistant to version 0.111.3
20-07-20 11:21:24 INFO (SyncWorker_5) [supervisor.docker.interface] Update image homeassistant/qemux86-64-homeassistant:0.112.4 to homeassistant/qemux86-64-homeassistant:0.111.3
20-07-20 11:21:24 INFO (SyncWorker_5) [supervisor.docker.interface] Pull image homeassistant/qemux86-64-homeassistant tag 0.111.3.
20-07-20 11:21:25 INFO (SyncWorker_5) [supervisor.docker.interface] Stop homeassistant application
20-07-20 11:21:29 INFO (SyncWorker_5) [supervisor.docker.interface] Clean homeassistant application
20-07-20 11:21:32 INFO (MainThread) [supervisor.homeassistant] Update pulse/client.config: /data/tmp/homeassistant_pulse
20-07-20 11:21:34 INFO (SyncWorker_0) [supervisor.docker.homeassistant] Start homeassistant homeassistant/qemux86-64-homeassistant with version 0.111.3
20-07-20 11:21:46 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation in progress
20-07-20 11:21:49 INFO (MainThread) [supervisor.auth] Auth request from core_mosquitto for iot
20-07-20 11:21:49 INFO (MainThread) [supervisor.auth] Home Assistant not running, check cache
20-07-20 11:21:49 INFO (MainThread) [supervisor.auth] Cache hit for iot
20-07-20 11:21:58 INFO (MainThread) [supervisor.homeassistant] Updated Home Assistant API token
20-07-20 11:22:23 INFO (MainThread) [supervisor.homeassistant] Home Assistant pip installation done
20-07-20 11:22:32 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
20-07-20 11:22:32 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_vscode
20-07-20 11:22:32 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
20-07-20 11:22:33 INFO (MainThread) [supervisor.homeassistant] Detect a running Home Assistant instance
20-07-20 11:22:33 INFO (MainThread) [supervisor.homeassistant] Successful run Home Assistant 0.111.3
20-07-20 11:22:34 INFO (SyncWorker_5) [supervisor.docker.interface] Cleanup images: ['homeassistant/qemux86-64-homeassistant:0.112.4']
`

**Comment by foxy82** [comment]
Removing all custom components (by renaming the dir from `custom_components` to `custom_components_bak` fixed this for me. 

I added them back in after the upgrade and it started up fine.

Although I removed a couple I no longer used - `braviatv_psk`, `alexa` (was used whilst I was waiting for my PR to make it to master), `dwains_theme`

**Comment by Toni981** [comment]
I did as you said but nothing .. does not go .. does not update .... 

Previously I found a temporary solution, I made a backup, I took a new sd and I loaded 0.112.4 and then uploaded the backup above .. did not give me error and loaded. 

Yesterday I tried to update it to version 0.113.0 and it doesn't update. 
I also tried to rename the custom_components folder but nothing .. how can I do? 

possible that only i have this problem?
